docs/adr/0007-modelling-history.md

```diff
@@ -0,0 +1,272 @@
+- Separate related data to different tables to reduce data usage
+
+### Won’t
+
+- Consider the sharing of data between translations, as per
+  [ADR-3](../adr/0003-initial-domain-modelling.md#decision)
+
+## Decision
+
+The proposed data model expands on the definitions of entities accordingly:
+
+**Document**: A record that represents all versions of a piece of content in a
+particular locale. It has many editions and many revisions. At any time it will
+have a current edition -  shown on Content Publisher index - and potentially a
+live edition which is currently on GOV.UK. The live and current edition can be
+the same. Document serves as a mutable entity that tracks when a Document was
+last edited and it is expected to be a joining point for document related data
+that is not associated with a particular editions.
+
+**Edition**: An edition is a numbered version of a document that has been, or is
+expected to be, published on GOV.UK. It is associated with a current revision
+and maintains an association with all revisions that have been associated with
+it. It has a current state (e.g. draft, submitted, published)
+and has knowledge of the past states it held. It is mutable and is expected
+as a place where any edition level database constraints can be placed.
+
+**Revision**: Represents the content of a document at a particular point in
+time. It has a number to indicate which revision of the document it is and
+stores who created it. Any time a user changes an aspect of a document a new
+revision is created and the current edition is updated to reference that
+revision - a revision cannot be edited and is considered immutable. A revision
+may have a association with a collection of items, such as images, when a new
+revision is created new images are not created but a new join between them is.
+
+This modelling is represented in the following diagram:
+
+![Class diagram](0007/class-diagram.png)
+
+Note: for brevity this diagram shows a reduced selection of fields and removes
+associations with Users.
+
+The purpose of models that not described above are:
+
+**State**: This represents the state of a document at a particular point in time,
+and is associated with an edition by being the current state. This model knows
+who created the state change and when. The model will likely have an polymorphic
+association to an object with any specifics of the state (such as time for
+scheduling, reason for retiring, etc). It is currently defined as immutable
+since state changes happen atomically but may well become mutable if it needs
+to track an asynchronous change in Publishing API.
+
+**Image**: Is an aspect of the current content of a document, it has a many to
+many relationship with a revision (since an image could be associated with
+many Revisions and a single Revision may have many images). It is immutable and
+any change to an image creates a new image object. We can identify an image
```

Comment:
When you say "changes to an image", do you mean users replacing an image? AFAIK the details of an image (e.g. alt-text, caption) can't be changed.

Reply:
I'm used to hearing the term 'aspect' describe a behaviour that cuts through many objects.  To me error handling and logging are examples but image makes me think of a shared object. Maybe just 'part of the current document'?

Reply:
> When you say "changes to an image", do you mean users replacing an image? AFAIK the details of an image (e.g. alt-text, caption) can't be changed/updated.

Those things can be updated, as can the crop.

Reply:
> Those things can be updated, as can the crop.

As long as the Document hasn't been published though, right?

Reply:
> > Those things can be updated, as can the crop.
> 
> As long as the Document hasn't been published though, right?

That's a current limitation that this work intends to resolve.

So in our no history world of now you can't update alt text / crop on a published image
When we have history you can edit alt text / crop etc on a published image